core/state/statedb.go

```diff
@@ -1060,11 +1075,19 @@ func (s *StateDB) PrepareAccessList(sender common.Address, dst *common.Address,
 	for _, addr := range precompiles {
 		s.AddAddressToAccessList(addr)
 	}
+
+	// Note: If an address is specified multiple times in the access list, only the
+	// last storage slots provided for it are used in predicates.
+	// During predicate verification, we enforce that a precompile address can only
+	// have one access tuple per address to avoid a situation where we verify multiple
+	// predicates and only expose the data from the last one.
+	s.predicateStorageSlots = make(map[common.Address][]byte)
```

Comment:
could we move this logic to a separate function eg `preparePredicateStorageSlots` or similar? Also we discussed switching to logic to consolidate all instances (although existing logic is fine too IMO)

Reply:
Yup